src/Tribe/iCal.php

```diff
@@ -197,48 +206,62 @@ public function maybe_add_link() {
 	 * Executes the iCal generator when the appropriate query_var or $_GET is setup.
 	 */
 	public function do_ical_template() {
-		// hijack to iCal template
-		if ( get_query_var( 'ical' ) || isset( $_GET['ical'] ) ) {
-			/**
-			 * Action fired before the creation of the feed is started, helpful to set up methods and other filters used
-			 * on this class.
-			 *
-			 * @since 4.6.11
-			 */
-			do_action( 'tribe_events_ical_before' );
+		// hijack to iCal template if query string included.
+		if (
+			! get_query_var( 'ical' )
+			&& ! isset( $_GET['ical'] )
+			&& ! get_query_var( 'outlook-ical' )
+			&& ! isset( $_GET['outlook-ical'] )
+		) {
+			return;
+		}
 
-			if ( ! $wp_query = tribe_get_global_query_object() ) {
-				return;
-			}
+		if (
+			get_query_var( 'outlook-ical' )
+			|| isset( $_GET['outlook-ical'] )
+		) {
+			$this->type = 'outlook';
+		}
+
+		/**
+		 * Action fired before the creation of the feed is started, helpful to set up methods and other filters used
+		 * on this class.
+		 *
+		 * @since 4.6.11
+		 */
+		do_action( 'tribe_events_ical_before' );
 
-			$event_ids = tribe_get_request_var( 'event_ids', false );
+		if ( ! $wp_query = tribe_get_global_query_object() ) {
+			return;
+		}
 
-			/**
-			 * Allows filtering the event IDs after the `Tribe__Events__ICal` class
-			 * tried to fetch them from the current request.
-			 *
-			 * @since 4.6.0
-			 *
-			 * @param array<int>|false Either a list of requested event post IDs or `false`
-			 *                         if the current request does not specify the event post
-			 *                         IDs to fetch.
-			 */
-			$event_ids = apply_filters( 'tribe_ical_template_event_ids', $event_ids );
+		$event_ids = tribe_get_request_var( 'event_ids', false );
 
-			if ( false !== $event_ids ) {
-				if ( empty( $event_ids ) ) {
-					die();
-				}
-				$event_ids = Arr::list_to_array( $event_ids );
-				$events = array_map( 'tribe_get_event', $event_ids );
-				$this->generate_ical_feed( $events );
-			} elseif ( is_singular( Tribe__Events__Main::POSTTYPE ) ) {
-				$this->generate_ical_feed( $wp_query->post );
-			} else {
-				$this->generate_ical_feed();
+		/**
+		 * Allows filtering the event IDs after the `Tribe__Events__ICal` class
+		 * tried to fetch them from the current request.
+		 *
+		 * @since 4.6.0
+		 *
+		 * @param array<int>|false Either a list of requested event post IDs or `false`
+		 *                         if the current request does not specify the event post
+		 *                         IDs to fetch.
+		 */
+		$event_ids = apply_filters( 'tribe_ical_template_event_ids', $event_ids );
```

Comment:
Namespace is wrong here, I think.

Reply:
This is an existing hook, I could add another, but I prefer not to change this one.